src/senselab/audio/tasks/features_extraction/praat_parselmouth.py

```diff
@@ -1,143 +1,1377 @@
+                window_size=window_length,
+                frame_shift=time_step,
+            )
+        )
+    if slope_tilt:
+        wf.add(
+            extract_slope_tilt_pt(
+                name="extract_slope_tilt_pt",
+                snd=wf.lzin.x,
+                floor=wf._extract_pitch_floor_pt.lzout.out,
+                ceiling=wf._extract_pitch_ceiling_pt.lzout.out,
+            )
+        )
+    if cpp_descriptors:
+        wf.add(
+            extract_cpp_descriptors_pt(
+                name="extract_cpp_descriptors_pt",
+                snd=wf.lzin.x,
+                floor=wf._extract_pitch_floor_pt.lzout.out,
+                ceiling=wf._extract_pitch_ceiling_pt.lzout.out,
+                frame_shift=time_step,
+            )
+        )
+    if duration:
+        wf.add(extract_audio_duration_pt(name="extract_audio_duration_pt", snd=wf.lzin.x))
+    if jitter:
+        wf.add(
+            extract_jitter_pt(
+                name="extract_jitter_pt",
+                snd=wf.lzin.x,
+                floor=wf._extract_pitch_floor_pt.lzout.out,
+                ceiling=wf._extract_pitch_ceiling_pt.lzout.out,
+            )
+        )
+    if shimmer:
+        wf.add(
+            extract_shimmer_pt(
+                name="extract_shimmer_pt",
+                snd=wf.lzin.x,
+                floor=wf._extract_pitch_floor_pt.lzout.out,
+                ceiling=wf._extract_pitch_ceiling_pt.lzout.out,
+            )
+        )
+
+    # setting multiple workflow outputs
+    output_connections = [("pitch_values_out", wf.extract_pitch_values_pt.lzout.out)]
+    if speech_rate:
+        output_connections.append(("speech_rate_out", wf.extract_speech_rate_pt.lzout.out))
+    if pitch:
+        output_connections.append(("pitch_out", wf.extract_pitch_descriptors_pt.lzout.out))
+    if intensity_descriptors:
+        output_connections.append(("intensity_out", wf.extract_intensity_descriptors_pt.lzout.out))
+    if harmonicity_descriptors:
+        output_connections.append(("harmonicity_out", wf.extract_harmonicity_descriptors_pt.lzout.out))
+    if formants:
+        output_connections.append(("formants_out", wf.measure_formants_pt.lzout.out))
+    if spectral_moments:
+        output_connections.append(("spectral_moments_out", wf.extract_spectral_moments_pt.lzout.out))
+    if slope_tilt:
+        output_connections.append(("slope_tilt_out", wf.extract_slope_tilt_pt.lzout.out))
+    if cpp_descriptors:
+        output_connections.append(("cpp_out", wf.extract_cpp_descriptors_pt.lzout.out))
+    if duration:
+        output_connections.append(("audio_duration", wf.extract_audio_duration_pt.lzout.out))
+    if jitter:
+        output_connections.append(("jitter_out", wf.extract_jitter_pt.lzout.out))
+    if shimmer:
+        output_connections.append(("shimmer_out", wf.extract_shimmer_pt.lzout.out))
+    wf.set_output(output_connections)
+
+    with pydra.Submitter(plugin=plugin) as sub:
```

Comment:
wherever we use this, we should also have `plugin_args`. in some settings leaving it to the default number of cores causes slowdowns if other tasks are using the same cores.